lib/Data/Printer/Filter/SCALAR.pm

```diff
@@ -21,10 +21,32 @@ sub _parse_scalar {
     my $ret;
 
     my $value = ref $scalar_ref ? $$scalar_ref : $scalar_ref;
+    my $numified = do { no warnings 'numeric'; 0+ $value };
```

Comment:
- I don't like $numified unconditionally calculated and later used only under if($dualvar) branch
- It's probably better to do 'my $t = $value; 0+$t' instead to prevent caching numerical value for non-numbers

Reply:
@dur-randir thanks for the review! Could you provide a test case where the dualvar check fails due to us not having the `my $t = $value` bit?

Reply:
That's not a failure case. My idea was to preserve as much of the original variable structure (in the original variable). But after further reading I see that it's already copied into $value, so that's unnecessary.

PS: I can't really test some other ideas about this patch as master branch fails for me in different ways. I've reported one of them, but should I report others? Or is it still in development, so I should wait until it stabilizes?